src/tunserver.cpp

```diff
@@ -1064,24 +1073,50 @@ string c_tunserver::rpc_sending_test(const string &input_json) {
 	ret["time_ms"] = time_ms;
 	ret["speed_mbps"] = ((packet_size * packet_count) / (time_ms / 1000.)) / (1024. * 1024.);
 	ret["msg"] = "ok:";
+	ret["state"] = "ok";
 	return ret.dump();
 }
 
-string c_tunserver::rpc_add_peer(const string &input_json)
-{
+string c_tunserver::rpc_add_peer(const string &input_json) {
 	auto input = nlohmann::json::parse(input_json);
 	auto peer = input["peer"].get<std::string>();
 	auto format = input["format"].get<std::string>();
 	nlohmann::json ret;
 	ret["cmd"] = "add_peer";
-	try{
-		if (format == "0.1")
-			add_peer_simplestring(peer);
-		else if (format == "1.0")
-			add_peer_simplestring_new_format(peer);
-		ret["msg"] = "ok: Peer added";
+	auto can_i_add_peer = [this](const c_haship_addr &hip) {
+		if (m_unban_if_banned) return true; // I can always add peer
```

Comment:
Instead, RPC json should say whether we should unban or not unban in this situation.

Also why is that flag atomic anyway, if anything wouldn't it be better to have a mutex for some "global configuration of RPC behaviour".